### ISSUE TITLE
(Master Bug) Segfault in rdimport (libtag.so)

### ISSUE THREAD

**Issue by deltecent** [ElvishArtisan/rivendell#337]
Getting a segfault importing an MP3. Will pursue when I have more time.

```Dec 26 19:34:16 maxsrv kernel: rdimport[25810]: segfault at 7f5fffffffe8 ip 00007f60258e5ad0 sp 00007ffd43a56d00 error 4 in libtag.so.1.12.0[7f6025863000+e7000]```

**Comment by deltecent** [comment]
```
*** Error in `/usr/bin/rdimport': double free or corruption (out): 0x00007f321f058c50 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81499)[0x7f321e50b499]
/lib64/libtag.so.1(_ZNK6TagLib6String9toCStringEb+0x2d)[0x7f322315ed6d]
/lib64/librd-3.0.0alpha04.so(_ZN10RDWaveFile15ReadId3MetadataEv+0x1044)[0x7f3223bafc64]
/lib64/librd-3.0.0alpha04.so(_ZN10RDWaveFile8openWaveEP10RDWaveData+0x3af)[0x7f3223bb0a1f]
/usr/bin/rdimport[0x40b779]
/usr/bin/rdimport[0x40f1aa]
/usr/bin/rdimport[0x40f3bb]
/usr/bin/rdimport[0x40f688]
/lib64/libQtCore.so.4(_ZN11QMetaObject8activateEP7QObjectPKS_iPPv+0x2ea)[0x7f321f6f833a]
/lib64/librd-3.0.0alpha04.so(_ZN13RDApplication15userChangedDataEv+0x43)[0x7f322395c203]
/lib64/libQtCore.so.4(_ZN11QMetaObject8activateEP7QObjectPKS_iPPv+0x2ea)[0x7f321f6f833a]
/lib64/librd-3.0.0alpha04.so(_ZN6RDRipc15DispatchCommandEv+0xaa4)[0x7f3223aff4e4]
/lib64/librd-3.0.0alpha04.so(_ZN6RDRipc9readyDataEv+0x158)[0x7f3223b00e98]
/lib64/libQtCore.so.4(_ZN11QMetaObject8activateEP7QObjectPKS_iPPv+0x2ea)[0x7f321f6f833a]
/lib64/libQtNetwork.so.4(+0xcaded)[0x7f321fd55ded]
/lib64/libQtNetwork.so.4(+0xd436d)[0x7f321fd5f36d]
/lib64/libQtGui.so.4(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x8c)[0x7f322019c10c]
/lib64/libQtGui.so.4(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x255)[0x7f32201a29e5]
/lib64/libQtCore.so.4(_ZN16QCoreApplication14notifyInternalEP7QObjectP6QEvent+0x8d)[0x7f321f6e38ad]
/lib64/libQtCore.so.4(+0x1b61a8)[0x7f321f7131a8]
/lib64/libglib-2.0.so.0(g_main_context_dispatch+0x159)[0x7f321aeeb969]
/lib64/libglib-2.0.so.0(+0x4acc8)[0x7f321aeebcc8]
/lib64/libglib-2.0.so.0(g_main_context_iteration+0x2c)[0x7f321aeebd7c]
/lib64/libQtCore.so.4(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x7e)[0x7f321f7123de]
/lib64/libQtCore.so.4(_ZN10QEventLoop13processEventsE6QFlagsINS_17ProcessEventsFlagEE+0x3f)[0x7f321f6e233f]
/lib64/libQtCore.so.4(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0x19d)[0x7f321f6e268d]
/lib64/libQtCore.so.4(_ZN16QCoreApplication4execEv+0x99)[0x7f321f6e7d59]
/usr/bin/rdimport[0x407370]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f321e4ac445]
/usr/bin/rdimport[0x4073d7]
```

**Comment by deltecent** [comment]
There is something about the COMPOSER tag in this MP3 that is blowing up TagLib:

```
[rd@maxsrv rdimport]$ .libs/rdimport --add-scheduler-code=DWN --normalization-level=-11 --autotrim-level=-30 --delete-source --startdate-offset=0 --enddate-offset=0 --log-filename=dropbox.log --log-directory=/home/rd --verbose MUSIC /home/rd/Downloads/*.mp3
/home/rd/Downloads/04 Wonderful Christmastime.mp3
rdwavefile.cpp:3513 ALBUM isNull=0 isEmpty=0 size=42 start
rdwavefile.cpp:3565 ALBUM end
rdwavefile.cpp:3513 ARTIST isNull=0 isEmpty=0 size=15 start
rdwavefile.cpp:3565 ARTIST end
rdwavefile.cpp:3513 COMMENT isNull=0 isEmpty=1 size=0 start
rdwavefile.cpp:3565 COMMENT end
rdwavefile.cpp:3513 COMPOSER isNull=0 isEmpty=0 size=-2145447937 start
```
When I view this MP3 with ```id3info``` the ```COMPOSER``` tag looks empty.

**Comment by deltecent** [comment]
This MP3 stores "empty" tags as a single NULL character, so for example, TCOM is stored as:

```TCOM   0x00 0x00 0x00 0x01   0x00 0x00   0x00```

This seems perfectly valid to me, but blows up TagLib.

**Comment by ElvishArtisan** [comment]
Fixed in [master 34f970e].
